doc/dhcp_server/port_based_dhcp_server_high_level_design.md

```diff
@@ -0,0 +1,1045 @@
+# IPv4 Port Based DHCP_SERVER in SONiC
+# High Level Design Document
+**Rev 0.1**
+
+# Table of Contents
+<!-- TOC -->
+
+- [IPv4 Port Based DHCP_SERVER in SONiC](#ipv4-port-based-dhcp_server-in-sonic)
+- [High Level Design Document](#high-level-design-document)
+- [Table of Contents](#table-of-contents)
+- [Revision](#revision)
+- [About this Manual](#about-this-manual)
+- [Scope](#scope)
+- [Definitions/Abbreviations](#definitionsabbreviations)
+                    - [Table 1: Abbreviations](#table-1-abbreviations)
+                    - [Table 2: Definitions](#table-2-definitions)
+- [Overview](#overview)
+    - [Background](#background)
+    - [Functional Requirements](#functional-requirements)
+    - [Configuration and Management Requirements](#configuration-and-management-requirements)
+- [Design](#design)
+    - [Design Overview](#design-overview)
+    - [Container](#container)
+    - [DHCP Server Daemon](#dhcp-server-daemon)
```

Comment:
This HLD wouldn't replace the original dhcp relay container functionality. In the scenario of disabling dhcp_server feature, functionality of dhcp_relay is exactly the same as before. Currently only support enabling per switch, means functionality of dhcp_relay cannot work with functionality dhcp_server. We will add restrictions in CLI to prevent users from configuring dhcp_relay when dhcp_server feature is enabled. Maybe add support in future releases for enabling dhcp_server feature per vlan, means enable dhcp_server in one vlan and enable dhcp_relay in other vlan.